awx_collection/plugins/module_utils/controller_api.py

```diff
@@ -339,6 +339,47 @@ def head_endpoint(self, endpoint, *args, **kwargs):
     def get_endpoint(self, endpoint, *args, **kwargs):
         return self.make_request('GET', endpoint, **kwargs)
 
+    def get_options_endpoint(self, endpoint, *args, **kwargs):
+        return self.make_request('OPTIONS', endpoint, **kwargs)['json']['actions']['POST']
+
+    def get_enforced_defaults(self, endpoint, *args, **kwargs):
+        endpoint_defaults = self.get_options_endpoint(endpoint)
+
+        default_fields = {
+            'organizations': {
+                'new_fields': {
+                    'description': '' if endpoint != 'organizations' else endpoint_defaults['description']['default'],
+                    'default_environment': '',
+                    'custom_virtualenv': '',
+                    'max_hosts': 0 if endpoint != 'organizations' else endpoint_defaults['max_hosts']['default'],
```

Comment:
The associations might have to be manually created for now but `new_fields` we should be able to get from `endpoint_defaults` right? i.e. something like: 
```
for option_name in response.json['POST']:
    try:
        new_fields['optoins_name'] = respose.json['actions']['POST'][option_name].get('default', 'awx_collection_undefined')
    except:
        raise Exception(f'{option_name} for model {endpoint} does not have a default!')
```
We might need to do a little bit of digging to make sure all defaults have been defined. Or if not, if there is a general default we can use like `""`.

Associations I'd like to do programatic too but I need to think about how we can accomplish that. Maybe we need the actual modules to construct that since the options page currently does not give us that information.

Reply:
The if else, is because, if you use the try except on them all, and you call job templates, it doesn't return information about schedules, it will except, and exit

By using if else, it will pull from api, for only the endpoint passed.

If we had a source of truth for this on the api, generated, with all endpoints populated, we could just pull in that structure

I looked at adding additional fields, or do something like it for each endpoint, but wasn't able to suss out how we did it.